kucher/view/main_window/register_view_widget/import_export_dialog.py

```diff
@@ -0,0 +1,302 @@
+    CheckResult.OUTSIDE_RANGE:       'This parameter value is outside permitted range: ',
+    CheckResult.UNKNOWN:             'Reason: ',
+}
+
+
+def export_registers(parent, registers):
+    """
+    Store all mutable and persistent registers in a .yml file
+    """
+    dialog_box = QWidget()
+    dialog_box.setGeometry(10, 10, 1000, 700)
+
+    file_name, _ = QFileDialog.getSaveFileName(dialog_box, 'Export configuration file', 'config.yml',
+                                               'YAML Files (*.yml  *.yaml);;All Files (*)')
+    if not file_name:
+        return
+    try:
+        _file = open(file_name, 'w+')
+    except Exception as ex:
+        show_error('Export failed',
+                   f'Cannot open {file_name}',
+                   f'Error: {str(ex)}',
+                   parent)
+        _logger.exception(f'File {file_name} could not be open: {str(ex)}')
+        return
+
+    async def executor():
+        try:
+            register_yaml = {}
+            for reg in registers:
+                if reg.mutable and reg.persistent:
+                    try:
+                        register_yaml[reg.name] = await reg.read_through()
+                    except Exception as ex:
+                        show_error('Export failed',
+                                   f'Parameter {reg.name} cannot be read.',
+                                   str(ex),
+                                   parent)
+                        _logger.exception(f'Register {reg} could not be read')
+                        return
+
+            yaml.dump(register_yaml, _file)
+            display_sucess_message('Export successful',
+                                   f'Parameters have been successfully exported to:\n{file_name}',
+                                   parent)
+        finally:
+            _file.close()
+
+    asyncio.get_event_loop().create_task(executor())
+
+
+def import_registers(parent, registers):
+    """
+    Import registers values from a .yml (or .yaml) file.
+    The file must match the following pattern:
+
+    full_register_name:
+    - register_value
+
+    e.g:
+
+    ctl.num_attempts:
+    - 100
+    ctl.pwron_slftst:
+    - false
+    ctl.spinup_durat:
+    - 1.7000000476837158
+
+    It can only contain available registers on the current device, with values within the allowed range. The values
+    must be the right type and dimension. If a register is not mentioned in the file, it will keep its former value.
+    """
+    dialog_box = QWidget()
+    dialog_box.setGeometry(10, 10, 1000, 700)
+    file_name, _ = QFileDialog.getOpenFileName(dialog_box, 'Import configuration file', '',
+                                               'YAML Files (*.yml *.yaml);;All Files (*)')
+    if not file_name:
+        return
+    try:
+        with open(file_name, 'r') as file:
+            imported_registers = yaml.load(file, Loader=yaml.Loader)
+            file.close()
```

Comment:
The point of the context manager is that it will close the file automatically when leaving the scope (like `finally`), so this is not needed.
```suggestion
```